### PR TITLE
Cotco 480 add pid be

### PR DESCRIPTION
**Pull request by peppe-hmrc**

### CODE REVIEW

**Comment by peppe-hmrc** [comment]
I would like you to review the use cases the query is covering and reviewing that I'm not missing any case or creating any inconsistent state

File: app/uk/gov/hmrc/ctc/repositories/TransitMongoQueryHelper.scala [hunk 1]
```diff
@@ -87,4 +94,31 @@ trait TransitMongoQueryHelper {
...
+      "$or" -> Seq(
+        Json.obj(selectReady()),
+        Json.obj(fields =
+          "$and" -> Seq(
+            Json.obj(selectProviderId(providerId)),
+            Json.obj("processingInfo.status" -> InProgress.toString)
+          )),
+        Json.obj(fields =
+          "$and" -> Seq(
+            Json.obj("processingInfo.providerId" -> Json.obj("$not" -> Json.obj("$eq" -> providerId))),
```
Comment (robertbutacu) [inline app/uk/gov/hmrc/ctc/repositories/TransitMongoQueryHelper.scala]: Why is this needed? Isnt it the branch where the record is expired?

File: app/uk/gov/hmrc/ctc/repositories/TransitRepository.scala [hunk 2]
```diff
@@ -193,15 +179,15 @@ class TransitRepository @Inject()(mongo: ReactiveMongoComponent)(implicit execut
     }.collect[List](Int.MaxValue, Cursor.FailOnError())
   }
 
-  def complete(id: String, record: OfficeOfTransit): ServiceResponse[Transit] = {
+  def complete(id: String, providerId: String, record: OfficeOfTransit): ServiceResponse[Transit] = {
```
Comment (robertbutacu) [inline app/uk/gov/hmrc/ctc/repositories/TransitRepository.scala]: is the providerId being used anywhere here?

Comment (robertbutacu) [reply app/uk/gov/hmrc/ctc/repositories/TransitRepository.scala]: Also, just to be pedantic, it would be better to have an "$unset" on the expiration time - thats useless once the record's processed

Comment (peppe-hmrc) [reply app/uk/gov/hmrc/ctc/repositories/TransitRepository.scala]: yes sorry, changing the complete is not part of this PR

File: app/uk/gov/hmrc/ctc/repositories/TransitMongoQueryHelper.scala (hunk 1 above)
Comment (peppe-hmrc) [reply app/uk/gov/hmrc/ctc/repositories/TransitMongoQueryHelper.scala]: good spot I think you are right ... gonna change it
